demos/ASP.Net Core/ITOps.ViewModelComposition.Gateway/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using System;
+using System.Threading.Tasks;
+
+namespace ITOps.ViewModelComposition.Gateway
+{
+    public static class ApplicationBuilderExtensions
+    {
+        public static void RunCompositionGateway(this IApplicationBuilder app, Action<IRouteBuilder> routes = null)
+        {
+            var routeBuilder = new RouteBuilder(app);
+            routes?.Invoke(routeBuilder);
+
+            app.UseRouter(routeBuilder.Build());
+        }
+
+        public static void RunCompositionGatewayWithDefaultRoutes(this IApplicationBuilder app)
+        {
+            app.RunCompositionGateway(routes =>
+            {
+                routes.MapComposableGet( template: "{controller}/{id:int?}");
+                routes.MapRoute("{*NotFound}", context =>
+                 {
+                     context.Response.StatusCode = StatusCodes.Status404NotFound;
+                     return Task.CompletedTask;
+                 });
+            });
+        }
```

Comment:
I was thinking less intrusive, like this:

```c#
public static void RunCompositionGatewayWithDefaultRoutes(this IApplicationBuilder app) =>
    app.RunCompositionGateway(routes =>
    {
        routes.MapComposableGet(template: "{controller}/{id:int?}");
        routes.MapRoute("*", context =>
         {
             context.Response.StatusCode = StatusCodes.Status404NotFound;
             return Task.CompletedTask;
         });
    });
```

I.e. the braces disappear and `=>` is appended to the declaration. Everything else stays the same, even the method body indentation.

BTW - I just noticed that there is an extra space before each of the braces that surrounds the handler delegate that's passed to the `MapRoute` method.